### DISCUSSION TITLE
bug in mermaid diagram support

### DISCUSSION THREAD

**Discussion by liza-unito** [community/community#84979] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

Hello, I was working on a embedding a mermaid diagram in some docs and noticed the following (probably undesirable?) behavior.
What a node contains the phrase `call function` preview crashes.

Example, here's a simple diagram
```mermaid
graph LR
A[text A]-->B(text B)
```

```
graph LR
A[text A]-->B(text B)
```

let's modify first text to say "call" and the second to be "function"

```mermaid
graph LR
A[call]-->B(function)
```

```
graph LR
A[call]-->B(function)
```

now, let's make the text "call function" for the second box - crashes.

```mermaid
graph LR
A[text A]-->B(call function)
```

note, the same will also happen if `call function` appears as part of the text  (which made this tricky to identify when initially debuggin)

```mermaid
graph LR
A[text A]-->B(sample text: call function B)
```

it's not an uncommon phrase to include in flow charts, so feels like the behavior is not intended. Probably has to do with HTML processor? I have not looked into trying to identify the root cause.

Best,
~Liza

**Comment by jkillingsworth** [comment, marked as answer]
It looks like the word `call` followed by a space is what's causing it.

```mermaid
graph LR
A[text A]-->B(call )
```

It also happens with the word `click` followed by a space.

```mermaid
graph LR
A[text A]-->B(click )
```

You can put a phrase such as `call function` in double quotes to get the desired result.

```mermaid
graph LR
A[text A]-->B("call function")
```
```
graph LR
A[text A]-->B("call function")
```
Check out the **Interaction** and **Special characters that break syntax** sections in the documentation.

* https://mermaid.js.org/syntax/flowchart.html#interaction
* https://mermaid.js.org/syntax/flowchart.html#special-characters-that-break-syntax

It looks like `call` and `click` are special keywords used for binding click events to a callback. There could be others. Who knows? It might be a good practice to use double quotes for all labels to avoid special words and characters that cause errors.

**Reply by liza-unito** [reply]
thanks very much, appreciated!
